editor/scene_tree_dock.cpp

```diff
@@ -468,6 +468,41 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 			script_create_dialog->config(inherits, path);
 			script_create_dialog->popup_centered();
 
+		} break;
+		case TOOL_ATTACH_SCRIPT: {
```

Comment:
There's a lot of duplicate code here, it would be better as a dedicated function for both attach/extend and to use a boolean to enable extending or not.